comp/core/secrets/secretsimpl/fetch_secret_test.go

```diff
@@ -185,59 +188,79 @@ func TestExecCommandError(t *testing.T) {
 	})
 }
 
-func TestFetchSecretExeceError(t *testing.T) {
+func TestFetchSecretExecError(t *testing.T) {
 	tel := fxutil.Test[telemetry.Component](t, nooptelemetry.Module())
 	resolver := newEnabledSecretResolver(tel)
 	resolver.commandHookFunc = func(string) ([]byte, error) { return nil, fmt.Errorf("some error") }
 	_, err := resolver.fetchSecret([]string{"handle1", "handle2"})
 	assert.NotNil(t, err)
 }
 
+func checkErrorCountMetric(t *testing.T, tel telemetry.Mock, expected int, errorKind, handle string) {
```

Comment:
Nit: Would you consider moving the utility function below the test that used it? I tend to write utility function below where they are called I tend to read top to bottom and I find it useful.

Reply:
Sure